src/PrestaShopBundle/Controller/Admin/SpecificPriceController.php

```diff
@@ -96,6 +101,78 @@ public function addAction(Request $request)
         return $response;
     }
 
+    /**
+     * Get one specific price list for a product
+     *
+     * @Template("@PrestaShop/Admin/Product/ProductPage/Forms/form_specific_price.html.twig")
+     *
+     * @param int $idSpecificPrice
+     *
+     * @return array
+     */
+    public function getUpdateFormAction($idSpecificPrice)
+    {
+        /** @var AdminProductWrapper $adminProductWrapper */
+        $adminProductWrapper = $this->get('prestashop.adapter.admin.wrapper.product');
+        try {
+            $price = $adminProductWrapper->getSpecificPriceDataById($idSpecificPrice);
+        } catch (\PrestaShopObjectNotFoundException $e) {
+            $message = $this->trans('Cannot find specific price %*price', 'Admin.Catalog.Notification', ['price' => $idSpecificPrice]);
+            return new Response($message, Response::HTTP_BAD_REQUEST);
+        }
+        $formData = $this->formatSpecificPriceToPrefillForm($idSpecificPrice, $price);
+
+        $options = [
+            'id_product' => $price->id_product,
+            'selected_product_attribute' => $price->id_product_attribute
+        ];
+
+        $formBuilder = $this->createFormBuilder();
+        $formBuilder->add('modal', SpecificPriceFormType::class, $options);
+
+        $form = $formBuilder->getForm();
+        $form->setData($formData);
+
+        $productAdapter = $this->get('prestashop.adapter.data_provider.product');
+        $product = $productAdapter->getProduct((int)$price->id_product);
+
+        return [
+            'form' => $form->createView()->offsetGet('modal'),
+            'has_combinations' => ($product->hasCombinations()),
+            'is_modal' => true,
+        ];
+    }
+
+    /**
+     * Update specific price Form process
+     *
+     * @param int idSpecificPrice
+     * @param Request $request
+     *
+     * @return string
+     */
+    public function updateAction($idSpecificPrice, Request $request)
+    {
+        $response = new JsonResponse();
+
+        // @todo: check it works
+        $idProduct = isset($request->get('form')['id_product']) ? $request->get('form')['id_product'] : null;
```

Comment:
Better to use `$request->request->get` if you're in POST, and `$request->query->get` if you're in GET. don't mix both.